pennylane/operation.py

```diff
@@ -964,6 +964,77 @@ def kraus_matrices(self):
         return self._kraus_matrices(*self.parameters)
 
 
+class Projection(Operation):
+    """Base class for projection operations
+
+    As with :class:`~.Operation`, the following class attributes must be
+    defined for all projections:
+
+    * :attr:`~.Operator.num_params`
+    * :attr:`~.Operator.num_wires`
+    * :attr:`~.Operator.par_domain`
+
+    To define a projection, the following attribute of :class:`~.Projection`
+    can be used to list the corresponding projectors.
+
+    * :attr:`~.Projection._projectors`
+
+    Args:
+        params (tuple[float, int, array, Variable]): operation parameters
+
+    Keyword Args:
+        wires (Sequence[int]): Subsystems the projection acts on. If not given, args[-1]
+            is interpreted as wires.
+        do_queue (bool): Indicates whether the operation should be
+            immediately pushed into a :class:`BaseQNode` circuit queue.
+            This flag is useful if there is some reason to run an Operation
+            outside of a BaseQNode context.
+    """
+
+    # pylint: disable=abstract-method
+
+    @classmethod
+    def _projectors(cls, *params, wires):
+        r"""Projectors representing a projection operation, specified in the
+        computational basis.
+
+        This is a class method that should be defined for all
+        new projections. It returns the projectors representing
+        the projection in the computational basis.
```

Comment:
```suggestion
        new projections.
```
Thought this was a bit similar to above